src/common/websocket/types.ts

```diff
@@ -0,0 +1,25 @@
+import { TSocketEndpoints } from './api.calls.types'
+
+export type TSocketEndpointNames = keyof TSocketEndpoints
+
+export type TSocketSubscribableEndpointNames =
+    | KeysMatching<TSocketEndpoints, { request: { subscribe?: number } }>
+    | 'exchange_rates'
+
+export type TSocketResponse<T extends TSocketEndpointNames> = TSocketEndpoints[T]['response']
+
+export type TSocketResponseData<T extends TSocketEndpointNames> =
+    TSocketResponse<T>[T extends 'ticks' ? 'tick' : T]
+
+export type TSocketRequest<T extends TSocketEndpointNames> = TSocketEndpoints[T]['request']
+
+export type TSocketRequestCleaned<T extends TSocketEndpointNames> = Omit<
```

Comment:
@mohsen-deriv I did some improvements in the types [here](https://github.com/binary-com/deriv-app/pull/7845), Not sure if it makes any difference for `deriv-com` or not but maybe you can have a look 🙇🏻